__tests__/index.ts

```diff
@@ -80,6 +80,106 @@ describe('test suites', () => {
   );
 });
 
+describe('issue 58 - key and value being identical', () => {
```

Comment:
Why create a new test group ? 
I think it would be nice to have this in the tests of utils.ts respectively for `areMapsEqual` and `areSetsEqual`. The tests there describe the behavior of the methods, and adding cases with different inputs would be helpful in understanding their requirements.
I like that it mentions the issue, so we understand why it was added. If we move the tests to utils.ts we could keep it in a comment.

Reply:
I can move it to utils, but I'll have to think about ruling it up vs splitting into a separate test group. I like the simplicity and clarity of it, vs rolling into some abstraction.

Reply:
What do you mean by "rolling it up"? 
An approach with [`it.each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) can work.

Reply:
Sorry, I was slippery-sloping ... `it.each` works just fine. :+1: Updated.